java-client/src/test/java/org/opensearch/client/opensearch/integTest/AbstractIndicesClientIT.java

```diff
@@ -192,4 +198,23 @@ public void testGetNotExistingIndexAlias() throws Exception {
             assertEquals(ex.getMessage(), "Request failed: [string_error] " + "alias [alias_not_exists] missing");
         }
     }
+
+    @Test
+    public void createIndex_withFlatObject_IndexCreatesSucessfully() throws IOException {
+        InfoResponse info = javaClient().info();
+        String version = info.version().number();
+        if (version.contains("SNAPSHOT")) {
+            version = version.split("-")[0];
+        }
+        assumeTrue("Flat Object is supported after version 2.7.0 only", Version.fromString(version).onOrAfter(Version.fromString("2.7.0")));
+        try {
+            final CreateIndexRequest createIndexRequest = new CreateIndexRequest.Builder().index("flat-object-test")
+                .mappings(m -> m.properties("sample_flat_object", Property.of(p -> p.flatObject(new FlatObjectProperty.Builder().build()))))
```

Comment:
Could you please add some properties to the field to make sure those are supported? Thank you.

Reply:
Based on the docs I dont think it supports properties https://opensearch.org/docs/latest/field-types/supported-field-types/flat-object/#supported-queries
I'll need to verify that and if its the case remove them from the method signatures